### ISSUE TITLE
pre-commit setup fails

### ISSUE THREAD

**Issue by benthevining** [jorisroovers/gitlint#302]
Every time I try to use the pre-commit hook, I get the following output:
```
[INFO] Installing environment for https://github.com/jorisroovers/gitlint.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/Users/benvining/.asdf/installs/python/3.9.10/bin/python3', '-mvirtualenv', '/Users/benvining/.cache/pre-commit/reponndrc_t8/py_env-python3')
return code: 1
expected return code: 0
stdout: (none)
stderr:
    Traceback (most recent call last):
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/runpy.py", line 188, in _run_module_as_main
        mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/runpy.py", line 147, in _get_module_details
        return _get_module_details(pkg_main_name, error)
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/runpy.py", line 111, in _get_module_details
        __import__(pkg_name)
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/site-packages/virtualenv/__init__.py", line 3, in <module>
        from .run import cli_run, session_via_cli
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/site-packages/virtualenv/run/__init__.py", line 14, in <module>
        from .plugin.creators import CreatorSelector
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/site-packages/virtualenv/run/plugin/creators.py", line 6, in <module>
        from virtualenv.create.via_global_ref.builtin.builtin_way import VirtualenvBuiltin
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/site-packages/virtualenv/create/via_global_ref/builtin/builtin_way.py", line 7, in <module>
        from virtualenv.create.creator import Creator
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/site-packages/virtualenv/create/creator.py", line 15, in <module>
        from virtualenv.discovery.cached_py_info import LogCmd
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/site-packages/virtualenv/discovery/cached_py_info.py", line 23, in <module>
        _CACHE[Path(sys.executable)] = PythonInfo()
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/site-packages/virtualenv/discovery/py_info.py", line 86, in __init__
        self.distutils_install = {u(k): u(v) for k, v in self._distutils_install().items()}
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/site-packages/virtualenv/discovery/py_info.py", line 152, in _distutils_install
        d = dist.Distribution({"script_args": "--no-user-cfg"})  # conf files not parsed so they do not hijack paths
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/site-packages/setuptools/dist.py", line 475, in __init__
        for ep in metadata.entry_points(group='distutils.setup_keywords'):
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/site-packages/setuptools/_vendor/importlib_metadata/__init__.py", line 999, in entry_points
        return SelectableGroups.load(eps).select(**params)
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/site-packages/setuptools/_vendor/importlib_metadata/__init__.py", line 449, in load
        ordered = sorted(eps, key=by_group)
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/site-packages/setuptools/_vendor/importlib_metadata/__init__.py", line 996, in <genexpr>
        eps = itertools.chain.from_iterable(
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/site-packages/setuptools/_vendor/importlib_metadata/_itertools.py", line 16, in unique_everseen
        k = key(element)
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/site-packages/setuptools/_vendor/importlib_metadata/__init__.py", line 931, in _normalized_name
        return self._name_from_stem(stem) or super()._normalized_name
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/site-packages/setuptools/_vendor/importlib_metadata/__init__.py", line 600, in _normalized_name
        return Prepared.normalize(self.name)
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/site-packages/setuptools/_vendor/importlib_metadata/__init__.py", line 855, in normalize
        return re.sub(r"[-_.]+", "-", name).lower().replace('-', '_')
      File "/Users/benvining/.asdf/installs/python/3.9.10/lib/python3.9/re.py", line 210, in sub
        return _compile(pattern, flags).sub(repl, string, count)
    TypeError: expected string or bytes-like object
```

I have tried with every version from 0.17.0 back to 0.10.0, always the same result. Any ideas?

**Comment by sigmavirus24** [comment]
The issue is happening when precommit tries to create the virtual environment for gitlint. Something in your installation if python via asdf is broken. I would try reinstalling python 3.9 via asdf and see if that helps but this isn't anything we can help with

**Comment by benthevining** [comment]
Oh, ok. Thank you!
